dbms/src/IO/Compression/CompressionCodecDeltaFOR.cpp

```diff
@@ -234,80 +127,72 @@ void CompressionCodecDeltaFOR::doDecompressData(
             ErrorCodes::CANNOT_DECOMPRESS,
             "Cannot decompress DeltaFor-encoded data. File has wrong header");
 
-    if (uncompressed_size == 0)
+    if (unlikely(uncompressed_size == 0))
         return;
 
     UInt8 bytes_size = source[0];
-    if unlikely (uncompressed_size % bytes_size != 0)
-        throw Exception(
-            ErrorCodes::CANNOT_DECOMPRESS,
-            "uncompressed size {} is not aligned to {}",
-            uncompressed_size,
-            bytes_size);
+    auto data_type = magic_enum::enum_cast<CompressionDataType>(bytes_size);
+    RUNTIME_CHECK(data_type.has_value());
 
     UInt32 source_size_no_header = source_size - 1;
-    switch (bytes_size)
+    switch (data_type.value())
     {
-    case 1:
-        decompressData<UInt8>(&source[1], source_size_no_header, dest, uncompressed_size);
+    case CompressionDataType::Int8:
+        DB::Compression::deltaFORDecoding<UInt8>(&source[1], source_size_no_header, dest, uncompressed_size);
         break;
-    case 2:
-        decompressData<UInt16>(&source[1], source_size_no_header, dest, uncompressed_size);
+    case CompressionDataType::Int16:
+        DB::Compression::deltaFORDecoding<UInt16>(&source[1], source_size_no_header, dest, uncompressed_size);
         break;
-    case 4:
-        decompressData<UInt32>(&source[1], source_size_no_header, dest, uncompressed_size);
+    case CompressionDataType::Int32:
+        DB::Compression::deltaFORDecoding<UInt32>(&source[1], source_size_no_header, dest, uncompressed_size);
         break;
-    case 8:
-        decompressData<UInt64>(&source[1], source_size_no_header, dest, uncompressed_size);
+    case CompressionDataType::Int64:
+        DB::Compression::deltaFORDecoding<UInt64>(&source[1], source_size_no_header, dest, uncompressed_size);
         break;
```

Comment:
What is the difference between these lines and `ordinaryDeltaFORDecoding` in lines 178~195?

Reply:
`ordinaryDeltaFORDecoding` are used in microbenchmark.